# HG changeset patch
# User Rik <rik@octave.org>
# Date 1440713960 25200
#      Thu Aug 27 15:19:20 2015 -0700
# Node ID 4bef1880098ce3bc15bb074c1f73babc84351a4d
# Parent  16b9ec39ff46a42c28074466a050f3afbca03442
More specific warning messages when saving onCleanup variables.

* ov-oncleanup.cc (warn_save_load): Delete function
(save_ascii, save_binary, save_hdf5): Call warning() with message about inability to save onCleanup variables.
(load_ascii, load_binary, load_hdf5): Silently skip over instances of onCleanup
in file that were not truly saved.

diff --git a/libinterp/octave-value/ov-oncleanup.cc b/libinterp/octave-value/ov-oncleanup.cc
--- a/libinterp/octave-value/ov-oncleanup.cc
+++ b/libinterp/octave-value/ov-oncleanup.cc
@@ -106,66 +106,63 @@ octave_oncleanup::~octave_oncleanup (voi
 octave_scalar_map
 octave_oncleanup::scalar_map_value (void) const
 {
   octave_scalar_map retval;
   retval.setfield ("task", fcn);
   return retval;
 }
 
-static void
-warn_save_load (void)
-{
-  warning ("onCleanup: load and save not supported");
-}
-
 bool
 octave_oncleanup::save_ascii (std::ostream& /* os */)
 {
-  warn_save_load ();
+  warning ("save: unable to save onCleanup variables, skipping");
+
   return true;
 }
 
 bool
 octave_oncleanup::load_ascii (std::istream& /* is */)
 {
-  warn_save_load ();
+  // Silently skip object that was not saved
   return true;
 }
 
 bool
 octave_oncleanup::save_binary (std::ostream& /* os */,
                                bool& /* save_as_floats */)
 {
-  warn_save_load ();
+  warning ("save: unable to save onCleanup variables, skipping");
+
   return true;
 }
 
 bool
 octave_oncleanup::load_binary (std::istream& /* is */, bool /* swap */,
                                oct_mach_info::float_format /* fmt */)
 {
-  warn_save_load ();
+  // Silently skip object that was not saved
   return true;
 }
 
 bool
 octave_oncleanup::save_hdf5 (octave_hdf5_id /* loc_id */,
                              const char * /* name */,
                              bool /* save_as_floats */)
 {
-  warn_save_load ();
+  warning ("save: unable to save onCleanup variables, skipping");
+
   return true;
 }
 
 bool
 octave_oncleanup::load_hdf5 (octave_hdf5_id /* loc_id */,
                              const char * /* name */)
 {
-  warn_save_load ();
+  // Silently skip object that was not saved
   return true;
 }
 
 void
 octave_oncleanup::print (std::ostream& os, bool pr_as_read_syntax)
 {
   print_raw (os, pr_as_read_syntax);
   newline (os);
